src/main/java/no/unit/nva/cristin/projects/ProjectsWrapper.java

```diff
@@ -112,17 +126,30 @@ public void setHits(List<NvaProject> hits) {
     }
 
     /**
-     * Assigns value to some field values using supplied query parameters.
+     * Assigns value to some of the field values using supplied headers and query parameters.
      *
-     * @param queryParams the query params
-     * @return ProjectsWrapper object with some of the field values set using query parameters
+     * @param headers     the headers from response
+     * @param queryParams the query params from request
+     * @return ProjectsWrapper object with some of the field values set using the supplied parameters
+     * @throws BadRequestException if page requested is invalid
      */
-    public ProjectsWrapper usingQueryParams(Map<String, String> queryParams) {
+    public ProjectsWrapper usingHeadersAndQueryParams(HttpHeaders headers, Map<String, String> queryParams)
+        throws BadRequestException {
+
+        this.size = getSizeHeader(headers);
         this.id = idUriFromParams(queryParams);
-        this.firstRecord = indexOfFirstEntryInPageCalculatedFromParams(queryParams);
+        this.firstRecord = this.size > 0 ? indexOfFirstEntryInPageCalculatedFromParams(queryParams) : 0;
```

Comment:
can you create a constant for the second 0? because it in unclear if it is the first element of an array (sometimes the first element is the 0th element) of it indicates that the value is not set.